### PR TITLE
[Docs] READmes for DockerHub

### LINKED ISSUE #10623: Document Docker Images in DockerHub
### Describe your feature request

After https://github.com/ray-project/ray/pull/10596, I think our Docker pipeline is fairly stable now. It's time to document all the tags and how to use them in Dockerhub README page. A good example I can think of is https://hub.docker.com/r/tensorflow/tensorflow/

The most ideal workflow is that we write the README.md in this repository, and push the result to dockerhub via https://github.com/christian-korneck/docker-pushrm in master build so the description will always stay in sync.

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
Add DockerHub READMEs into GitHub. 

## Related issue number

Closes #10623

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by simon-mo** [comment]
This doesn't setup the automatic push to dockerhub yet, right? just confirming

**Comment by ijrsvt** [comment]
Correct, the next step is for me to see how easy it is to add that into CI

**Review by simon-mo — APPROVED** [review]
LGTM but let's wait for @edoakes to take another pass. Two high level:
- Maybe we should undocument base-deps and ray-deps in ray-ml and ray regular user don't need to see it. 
- Can we link the docker file for each image in their readme? that would be amazing for users.

File: docker/autoscaler/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## About
+This image used to be the base image for using the Ray autoscaler on a Kubernetes cluster. 
+
+It is identical to [`rayproject/ray-ml`](https://hub.docker.com/repository/docker/rayproject/ray-ml).
+
+Please switch to that image if you are still using this one :)
```
[the remark this answers is not in the document]
Comment (ijrsvt) [reply docker/autoscaler/README.md:6]: SGTM!

File: docker/base-deps/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+## About
+This is an internal iamge that has the system-level dependencies for `Ray` and the `Ray Autoscaler`. The `ray-deps` image is built on top of this. This image is built periodically or when dependencies are added.
```
[the remark this answers is not in the document]
Comment (ijrsvt) [reply docker/base-deps/README.md:2]: Oops--fixed!

File: docker/ray-ml/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+## About
+This image is an extension of the [`rayproject/ray`](https://hub.docker.com/repository/docker/rayproject/ray) image. It includes all extended requirements of `Tune`, `RLlib` and `SGD`. It is a well-provisioned starting point for trying out the Ray ecosystem.
```
[the remark this answers is not in the document]
Comment (ijrsvt) [reply docker/ray-ml/README.md:2]: switched to
`RLlib`, `Serve` and `Tune`.

**Comment by ijrsvt** [comment]
Dropped the links to `base/ray` `-deps` in `ray` & `ray-ml`.
Also will add links to READMEs :)

File: docker/autoscaler/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+## About
```
Comment (edoakes) [inline docker/autoscaler/README.md:1]: I would more explicitly mark this as deprecated and say we'll stop pushing it. You can just say `DEPRECATED. This image used to be the base image for the Ray autoscaler, but it has been replaced by rayproject/ray-ml. Please use that instead, this image will be removed in the near future.`

File: docker/base-deps/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+## About
```
Comment (edoakes) [inline docker/base-deps/README.md:1]: Should users use this at all? If not let's have a pointer to the images they should be using.

File: docker/ray/README.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+## About
+Ray is an open source framework that provides a simple, universal API for building distributed applications. Ray is packaged with RLlib, a scalable reinforcement learning library, and Tune, a scalable hyperparameter tuning library.
```
Comment (edoakes) [inline docker/ray/README.md:2]: I'd just link to the github page instead of writing an intro about ray here

**Review by edoakes — APPROVED** [review]
Looks good, just some small suggestions!

File: docker/autoscaler/README.md (hunk 4 above)
Comment (ijrsvt) [reply docker/autoscaler/README.md:1]: Sounds good!

File: docker/base-deps/README.md (hunk 5 above)
Comment (ijrsvt) [reply docker/base-deps/README.md:1]: Nope, I'll leave pointers here & on `ray-deps`

File: docker/ray/README.md (hunk 6 above)
Comment (ijrsvt) [reply docker/ray/README.md:2]: Done!

**Pull request merged**